app/src/shared/components/TextField/index.jsx

```diff
@@ -1,35 +1,41 @@
 // @flow
 
-import React from 'react'
+import React, { useCallback } from 'react'
 import cx from 'classnames'
 
+import NumberField from './NumberField'
 import styles from './textField.pcss'
 
 type Props = {
     className?: string,
     onAutoComplete?: (boolean) => void,
+    type?: string,
 }
 
-class TextField extends React.Component<Props> {
-    onAnimationStart = ({ animationName }: SyntheticAnimationEvent<EventTarget>) => {
-        const { onAutoComplete } = this.props
-
+const TextField = ({ className, onAutoComplete, ...props }: Props) => {
+    const onAnimationStart = useCallback(({ animationName }: SyntheticAnimationEvent<EventTarget>) => {
         if (onAutoComplete && (animationName === styles.onAutoFillStart || animationName === styles.onAutoFillCancel)) {
             onAutoComplete(animationName === styles.onAutoFillStart)
         }
-    }
-
-    render = () => {
-        const { className, onAutoComplete, ...props } = this.props
-
-        return (
-            <input
-                {...props}
-                className={cx(className, styles.root)}
-                onAnimationStart={this.onAnimationStart}
-            />
-        )
-    }
+    }, [onAutoComplete])
+
+    return (
+        <React.Fragment>
+            {props.type === 'number' ? (
+                <NumberField
+                    {...props}
+                    className={cx(className, styles.root)}
+                    onAnimationStart={onAnimationStart}
+                />
+            ) : (
+                <input
+                    {...props}
+                    className={cx(className, styles.root)}
+                    onAnimationStart={onAnimationStart}
+                />
+            )}
+        </React.Fragment>
+    )
```

Comment:
Does it really need a fragment? How 'bout this:

```jsx
return props.type === 'number' ? (
    // <NumberField …
) : (
    // <input …
)
```
or even
```jsx
const Tag = props.type === 'number' ? NumberField : 'input'

return (
    <Tag
        {...props}
        className={cx(className, styles.root)}
        onAnimationStart={onAnimationStart}
    />
)
```
Maybe something to consider. :)